tasks/indexing/task_indexing.go

```diff
@@ -248,6 +255,40 @@ func (i *IndexingTask) recordCompletion(ctx context.Context, task itask, taskID
 }
 
 func (i *IndexingTask) CanAccept(ids []harmonytask.TaskID, engine *harmonytask.TaskEngine) (*harmonytask.TaskID, error) {
+	ctx := context.Background()
+
+	indIDs := make([]int64, len(ids))
+	for x, id := range ids {
+		indIDs[x] = int64(id)
+	}
+	acceptables := map[harmonytask.TaskID]bool{}
+
+	for _, t := range ids {
+		acceptables[t] = true
+	}
+
+	var ac []struct {
+		TaskID      harmonytask.TaskID `db:"indexing_task_id"`
+		ShouldIndex bool               `db:"should_index"`
+	}
+
+	err := i.db.Select(ctx, &ac, `select indexing_task_id, should_index from market_mk12_deal_pipeline where indexing_task_id = ANY ($1)`, indIDs)
+	if err != nil {
+		return nil, xerrors.Errorf("getting pending indexing tasks: %w", err)
+	}
+
+	for _, t := range ac {
+		if _, ok := acceptables[t.TaskID]; !ok {
+			continue
+		}
+
+		// Accept any task which should not be indexed as
+		// it does not require storage access
```

Comment:
Technically true, but we really, really want to run indexing on the storage node. Running indexing remotely will use massive amount of bandwidth unnecessarily

Reply:
This logic is only for those task which should not be indexed. The reason is simple, if I said don't keep unsealed and there is no unsealed then indexing task will never be accepted. Thus allow these tasks to be picked up by any node and then do will simply mark them as finished and make the entry about the piece in the table. Since DO() is doing a few things we need, I am not putting the logic to mark complete in taskID creation function like for IPNI job.

Reply:
Oooh

Why is `ShouldIndex` not just a part of the `where` clause in the query?